Typescript/TypeBasic.md

```diff
@@ -0,0 +1,218 @@
+# Types
+
+**Types in Typescript**
+
+Typescript types is categorized into:
+• Primitive types
+• Object types
+
+**Primitive types**
+
+string
+number
+boolean
+null
+undefined
+symbol
+
+**Object types**
+
+Object types are functions, arrays, classes, etc.
+
+**Purposes of types in Typescript:**
+
+There are two main purposes of types in Typescript:
+• First, types are used by the Typescript compiler to analyze your code for errors
+• Second, types allow you to understand what values are associated with variables.
+
+**What is Type Annotation in Typescript**
+
+Typescript uses type annotations to explicitly specify types for identifiers such variables, functions, objects, etc.
+
+Typescript uses the syntax `: type` after an identifier as the type annotation, where type can be any valid type.
+
+`let variableName: type = value`;
+
+**Number**
+
+number is for numbers like 42. JavaScript does not have a special run time value for integers, so there’s no equivalent to int or float - everything is simply number
+
+**string**
+
+string represents string values like "Hello, world"
+`let title: string = "Web Developer"`;
+
+**Boolean**
+
+boolean is for the two values true and false
+`let pending: boolean`;
+`pending = true`;
+
```

Comment:
Please add other 3 types as well, undefined, null and symbol. Also mention during the discussion about bigint etc.